pkg/logs/auditor/auditor.go

```diff
@@ -220,10 +249,26 @@ func (a *Auditor) updateRegistry(identifier string, offset string, tailingMode s
 		// specially want to avoid storing the offset
 		return
 	}
-	a.registry[identifier] = &RegistryEntry{
-		LastUpdated: time.Now().UTC(),
-		Offset:      offset,
-		TailingMode: tailingMode,
+	entry, exists := a.registry[identifier]
+	if !(exists && entry.CurrentConfigID != configID) {
+		a.registry[identifier] = &RegistryEntry{
+			LastUpdated:     time.Now().UTC(),
+			Offset:          offset,
+			TailingMode:     tailingMode,
+			CurrentConfigID: configID,
+		}
+	}
+}
+
+// updateCurrentConfigID updates the registry entry matching identifier with new the offset and timestamp
+func (a *Auditor) updateCurrentConfigID(identifier, configID string) {
+	a.registryMutex.Lock()
+	defer a.registryMutex.Unlock()
```

Comment:
`updateCurrentConfigID` is called when receiving an id from a channel. `updateCurrentConfigID` uses a mutex. Why using 2 synchronisation mechanisms?

Reply:
I think there is two points here, being consistent wiht `updateRegistry` and protecting the registry itself as it can it's access by direct calls that uses the `readOnlyRegistryCopy`.